python/pyiceberg/expressions/visitors.py

```diff
@@ -89,33 +91,33 @@ def visit_and(self, left_result: T, right_result: T) -> T:
         """Visit method for an And boolean expression
 
         Args:
-            left_result (T): The result of visiting the left side of the expression
-            right_result (T): The result of visiting the right side of the expression
+            left_result (R): The result of visiting the left side of the expression
+            right_result (R): The result of visiting the right side of the expression
         """
 
     @abstractmethod
     def visit_or(self, left_result: T, right_result: T) -> T:
         """Visit method for an Or boolean expression
 
         Args:
-            left_result (T): The result of visiting the left side of the expression
-            right_result (T): The result of visiting the right side of the expression
+            left_result (R): The result of visiting the left side of the expression
+            right_result (R): The result of visiting the right side of the expression
         """
 
     @abstractmethod
-    def visit_unbound_predicate(self, predicate) -> T:
+    def visit_unbound_predicate(self, predicate: UnboundPredicate) -> T:
```

Comment:
Should be `UnboundPredicate[L]`, I think.

Reply:
This is now possible again, indeed. Thanks!